src/family/family.service.ts

```diff
@@ -18,4 +22,40 @@ export class FamilyService {
       },
     });
   }
+
+  async create(data: CreateFamilyInput): Promise<Family> {
+    const familySlug = slug(data.name);
+    const extendedData = {
+      ...data,
+      slug: familySlug,
+    };
+
+    // eslint-disable-next-line fp/no-let
+    let family: Family | null = null;
+    // eslint-disable-next-line fp/no-let
+    let counter = 0;
+
+    // eslint-disable-next-line fp/no-loops
+    while (!family) {
+      try {
+        if (counter !== 0) {
+          extendedData.slug = `${familySlug}-${counter}`;
+        }
+
+        // eslint-disable-next-line no-await-in-loop
+        family = await this.prisma.family.create({
+          data: extendedData,
+        });
+      } catch (error: any) {
+        if (error?.code === 'P2002' && error?.meta?.target?.includes('slug')) {
+          counter++;
+        } else {
+          this.logger.error({ errorOnCreateFamily: error });
+          throw error;
+        }
+      }
+    }
```

Comment:
As mentioned before, it's not good practice to have to `await` in the loop,
this solution is not appropriate,

The solution I can represent right now:
 1. make a find query to check proposed slug ( like a proposed slug ) is assigned before or not
 2. then based on the result decide to change `slug` to another one.

In this way, there is only 2 separate query, but no query in the loop.

Reply:
Done.

this way is not perfect but in our scale is totally perfect.

Reply:
But at least it's better than before!